parser/v2/types.go

```diff
@@ -375,17 +375,27 @@ func (t HTMLTemplate) Write(w io.Writer, indent int) error {
 type TrailingSpace string
 
 const (
-	SpaceNone       TrailingSpace = ""
-	SpaceHorizontal TrailingSpace = " "
-	SpaceVertical   TrailingSpace = "\n"
+	SpaceNone           TrailingSpace = ""
+	SpaceHorizontal     TrailingSpace = " "
+	SpaceVertical       TrailingSpace = "\n"
+	SpaceVerticalDouble TrailingSpace = "\n\n"
 )
 
 var ErrNonSpaceCharacter = errors.New("non space character found")
 
-func NewTrailingSpace(s string) (ts TrailingSpace, err error) {
+func NewTrailingSpace(s string, allowMulti bool) (ts TrailingSpace, err error) {
```

Comment:
This function is getting quite "indented", is there any refactors we can make here to make the flow easier to follow?

Reply:
Possibly could be a different function `NewTrailingSpace` and `NewMultilineTrailingSpace`. Rather than a flag. Would that potentially make it cleaner, or just add duplication?

Reply:
The majority of that function would be the same between the two, so you'd end up either duplicating the code, or finding a different type of abstraction. I'd say this solution is the least "abstraction-hell" and at this point I wouldn't worry too much with it.
But I agree if more parameters are introduced, its responsibility would be too large, but not really in its current state I think.